CONTRIBUTING.md

```diff
@@ -0,0 +1,114 @@
+# Contributing
+
+This document provides guidelines for contributing to this repository.
+
+## Development Environment Setup
+
+### Prerequisites
+* Linux machine with x86-64 architecture with bash installed.
+* Basic tools: curl, tar, make.
+
+### Running the Automated Tests
+1. Clone your fork of the project.
+2. Download dependencies and run automated tests: `make all`.
```

Comment:
Not sure if `Download dependencies` is clear enough

Reply:
Yes. That's the change. Can you elaborate on what's not clear? I'll try to rephrase accordingly.

Reply:
I don't think it's clear which dependencies should be downloaded. I would make it more clear.

Reply:
> Running make downloads them

Exactly. Therefore, I think writing `download additional dependencies` might imply that somethings needs to be downloaded manually except just running make.

Reply:
I think I understand the confusion now. See if the current wording makes more sense. I was trying to avoid putting the command at the beginning so people won't execute it without first reading what they are about to do, but alright.

Reply:
@eisraeli are you ok with closing this point?